lib/trace_location/collector.rb

```diff
@@ -16,8 +16,17 @@ def collect(match:, ignore:, &block)
         cache = {}
 
         tracer = TracePoint.new(:call, :return) do |trace_point|
-          next if match && !trace_point.path.to_s.match?(/#{match}/)
-          next if ignore && trace_point.path.to_s.match?(/#{ignore}/)
+          if match.is_a? Array
+            next if match.any? && match.none? { |content| trace_point.path.to_s.match? /#{content}/ }
+          else
+            next if match && !trace_point.path.to_s.match?(/#{match}/)
+          end
+
+          if ignore.is_a? Array
+            next if ignore.any? && ignore.any? { |content| trace_point.path.to_s.match? /#{content}/ }
+          else
+            next if ignore && trace_point.path.to_s.match?(/#{ignore}/)
+          end
```

Comment:
@shioimm  How do you feel converting value like `Array(value).join('|')` without `is_a?` ?

Reply:
That seems to be a good idea. I'll modify and send again.
Thank you!